pkg/scheduler/ugm/manager.go

```diff
@@ -0,0 +1,288 @@
+			}
+		}
+	} else {
+		log.Logger().Error("user tracker must be available in userTrackers map",
+			zap.String("user", user.User))
+		return fmt.Errorf("user tracker for %s is missing in userTrackers map", user.User)
+	}
+
+	group, err := m.getGroup(user)
+	if err != nil {
+		return err
+	}
+	groupTracker := m.groupTrackers[group]
+	if groupTracker != nil {
+		err := groupTracker.decreaseTrackedResource(queuePath, applicationID, usage, removeApp)
+		if err != nil {
+			log.Logger().Error("Problem in decreasing the group resource usage",
+				zap.String("user", user.User),
+				zap.String("group", group),
+				zap.String("queue path", queuePath),
+				zap.String("application", applicationID),
+				zap.String("resource", usage.String()),
+				zap.Bool("removeApp", removeApp),
+				zap.String("err message", err.Error()))
+			return err
+		}
+		if removeApp {
+			if m.isGroupRemovable(groupTracker) {
+				delete(m.groupTrackers, group)
+			}
+		}
+	} else {
+		log.Logger().Error("appGroupTrackers tracker must be available in groupTrackers map",
+			zap.String("appGroupTrackers", group))
+		return fmt.Errorf("appGroupTrackers tracker for %s is missing in groupTrackers map", group)
+	}
+	return nil
+}
+
+func (m *Manager) GetUserResources(user security.UserGroup) *resources.Resource {
+	return nil
+}
+
+func (m *Manager) GetGroupResources(group string) *resources.Resource {
+	return nil
+}
+
+func (m *Manager) GetUsersResources() []*UserTracker {
+	return nil
+}
+
+func (m *Manager) GetGroupsResources() []*GroupTracker {
+	return nil
+}
+
+func (m *Manager) ensureGroupTrackerForApp(queuePath string, applicationID string, user security.UserGroup) error {
+	userTracker := m.userTrackers[user.User]
+	if !userTracker.hasGroupForApp(applicationID) {
+		var groupTracker *GroupTracker
+		group, err := m.getGroup(user)
+		if err != nil {
+			return err
+		}
+		if m.groupTrackers[group] == nil {
+			log.Logger().Debug("Group tracker does not exist. Creating group tracker object and linking the same with application",
+				zap.String("application", applicationID),
+				zap.String("queue path", queuePath),
+				zap.String("user", user.User),
+				zap.String("group", group))
+			groupTracker = newGroupTracker(group)
+			m.groupTrackers[group] = groupTracker
+		} else {
+			log.Logger().Debug("Group tracker already exists and linking (reusing) the same with application",
+				zap.String("application", applicationID),
+				zap.String("queue path", queuePath),
+				zap.String("user", user.User),
+				zap.String("group", group))
+			groupTracker = m.groupTrackers[group]
+		}
+		userTracker.setGroupForApp(applicationID, groupTracker)
+	}
+	return nil
+}
+
+// getGroup Based on the current limitations, username and group name is same. Groups[0] is always set and same as username.
+// It would be changed in future based on user group resolution, limit configuration processing etc
+func (m *Manager) getGroup(user security.UserGroup) (string, error) {
```

Comment:
Question: what if, for whatever reason, we don't have groups for a certain user? Can that happen? Eg. we disable the user-group retrieval in the AC, what happens here?